bgpd/bgp_rpki.c

```diff
@@ -1326,15 +1326,18 @@ DEFPY (show_rpki_prefix_table,
 {
 	struct json_object *json = NULL;
 
+
+	if (uj)
+		json = json_object_new_object();
+
 	if (!is_synchronized()) {
-		if (!uj)
+		if (uj)
+			vty_json(vty, json);
+		else
 			vty_out(vty, "No connection to RPKI cache server.\n");
 		return CMD_WARNING;
 	}
 
-	if (uj)
-		json = json_object_new_object();
-
 	print_prefix_table(vty, json);
```

Comment:
We need to fix JSON output for `print_prefix_table()` too. This part (to return `{}`):

```c
	if (!group) {
		if (!json)
			vty_out(vty, "Cannot find a connected group.\n");
		return;
	}
```

Reply:
done